### PR TITLE
replace start pos with range

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Replace startPos with range in codeGen 
Related to #763

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/front/codeGen.bal [hunk 1]
```diff
@@ -500,7 +500,7 @@ function codeGenWhileStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environm
     }
     else {
         // condition is false and body is non-empty
-        return cx.semanticErr("unreachable code", stmt.body.stmts[0].startPos);
+        return cx.semanticErr("unreachable code", s:range(stmt.body));
```
Comment (jclark) [inline compiler/modules/front/codeGen.bal]: Why are you changing from `stmt.body.stmts[0]` to `stmt.body`? Don't make changes that are not part of the issue. This change is wrong.

File: compiler/modules/front/codeGen.bal [hunk 2]
```diff
@@ -1172,7 +1172,7 @@ function codeGenCallStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environme
         return check codeGenCheckingStmt(cx, startBlock, env, expr.checkingKeyword, expr.operand, expr.kwPos);
     }
     if result != () {
-        return cx.semanticErr("return type of function or method in call statement must be nil", stmt.startPos);
+        return cx.semanticErr("return type of function or method in call statement must be nil", s:range(stmt));
```
Comment (jclark) [inline compiler/modules/front/codeGen.bal]: Just do expressions to start with as I suggested: https://github.com/ballerina-platform/nballerina/issues/763#issuecomment-1003478084

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front/codeGen.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/front/codeGen.bal]: fixed

File: compiler/modules/front/codeGen.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply compiler/modules/front/codeGen.bal]: fixed

**Pull request merged**